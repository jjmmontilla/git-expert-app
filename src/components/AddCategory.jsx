import { PropTypes } from "prop-types";
import { useState } from "react";

export const AddCategory = ({onNewCategory}) => {
  const [inputVal, setInputVal] = useState('');

  const onInputChange = ({target}) => {
    setInputVal(target.value);
  } 

  const onSubmit = (event) => {
    event.preventDefault();
    const val = inputVal.trim();
    if (val.length <= 1) return;
    onNewCategory(val);
    setInputVal('');
  }

  return (
    <form onSubmit={onSubmit} aria-label="form">
      <div>
        <input type='text' 
          placeholder='Bucar Gifs' 
          value={inputVal} 
          onChange={onInputChange} />
      </div>
    </form>
  )
}

AddCategory.propTypes = {
  onNewCategory: PropTypes.func.isRequired,
}
