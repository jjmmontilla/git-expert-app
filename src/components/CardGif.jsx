import PropTypes from 'prop-types';

export const CardGif = ({url, title}) => {
  return (
    <div className="card">
      <img src={url} alt={title} width='100' />
      <p>{title}</p>
    </div>
  )
}

CardGif.propTypes = {
  url: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired
}