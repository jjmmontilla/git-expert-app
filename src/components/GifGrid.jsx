import PropTypes from 'prop-types';
import { useFetchGifs } from "../hooks/useFetchGifs";
import { CardGif } from "./cardGif";

export const GifGrid = ({category}) => {
  const { images, isLoading } = useFetchGifs(category);

  return (
    <>
      <h4>{ category }</h4>
      
      { isLoading && (<h2>Cargando..</h2>) }

      <div className="card-grid">
        {
          images.map((image) => (
          <CardGif key={image.id} {...image} />))
        }
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
