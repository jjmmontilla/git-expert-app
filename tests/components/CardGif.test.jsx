import { render, screen } from '@testing-library/react';
import { CardGif } from '../../src/components/CardGif';

describe('Pruebas en <CardGif />',  () => {
    const title = 'Dragon ball';
    const url = 'https://media3.giphy.com/media/VXJWhaO7afRe/giphygif&ct=g';

    test('Debe hacer match con el snatshop', () => {
        const { container } = render( <CardGif title={title} url={url} />);
        expect( container ).toMatchSnapshot();
    });

    test('Debe mostrar la img con el url y el Alt indicado', () => {
        render( <CardGif title={title} url={url} /> );
        const { src, alt } = screen.getByRole('img');
        expect(src).toBe(url);
        expect(alt).toBe(title);
    });

    test('Debe de mostrar el titulo en el componente', () => {
        render( <CardGif title={title} url={url} /> );
        expect(screen.getAllByText(title)).toBeTruthy();
    });
});