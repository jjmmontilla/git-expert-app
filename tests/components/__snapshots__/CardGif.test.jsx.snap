// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CardGif /> Debe hacer match con el snatshop 1`] = `
<div>
  <div
    class="card"
  >
    <img
      alt="Dragon ball"
      src="https://media3.giphy.com/media/VXJWhaO7afRe/giphygif&ct=g"
      width="100"
    />
    <p>
      Dragon ball
    </p>
  </div>
</div>
`;
